src/autoload.rs

```diff
@@ -336,10 +336,20 @@ fn test_autoload() {
     }
 
     fn touch_file(path: &wstr) {
-        let fd = wopen_cloexec(path, O_RDWR | O_CREAT, 0o666);
-        assert!(fd >= 0);
+        use nix::sys::stat::Mode;
+
+        let fd = wopen_cloexec(
+            path,
+            OFlag::O_RDWR | OFlag::O_CREAT,
+            Mode::S_IRUSR
+                | Mode::S_IWUSR
+                | Mode::S_IRGRP
+                | Mode::S_IWGRP
+                | Mode::S_IROTH
+                | Mode::S_IWOTH,
```

Comment:
Can you use just use `Mode::from_bits(0o666).unwrap()` here? Easier to grok for us greybeards.

Reply:
Done